pkg/gds/gds_test.go

```diff
@@ -264,118 +293,175 @@ func (s *gdsTestSuite) TestSearch() {
+		request := &api.SearchRequest{}
+		reply, err := client.Search(ctx, request)
+		require.NoError(err)
+		require.Empty(reply.Error)
+		require.Len(reply.Results, 0)
+	})
 
-	// Filter by country
-	request = &api.SearchRequest{
-		Name:    []string{"CharlieBank"},
-		Country: []string{charlieVASP.Entity.CountryOfRegistration},
-	}
-	reply, err = client.Search(ctx, request)
-	require.NoError(err)
-	require.Empty(reply.Error)
-	require.Len(reply.Results, 1)
-
-	// Filter by country - no results
-	request = &api.SearchRequest{
-		Name:    []string{"CharlieBank"},
-		Country: []string{"US"},
-	}
-	reply, err = client.Search(ctx, request)
-	require.NoError(err)
-	require.Empty(reply.Error)
-	require.Len(reply.Results, 0)
+	s.Run("ByName", func() {
+		// Search by name
+		require := s.Require()
+		request := &api.SearchRequest{
+			Name: []string{"Hotel Corp"},
+		}
 
-	// Filter by category
-	request = &api.SearchRequest{
-		Name:             []string{"CharlieBank"},
-		BusinessCategory: []pb.BusinessCategory{charlieVASP.BusinessCategory},
-	}
-	reply, err = client.Search(ctx, request)
-	require.NoError(err)
-	require.Empty(reply.Error)
-	require.Len(reply.Results, 1)
+		reply, err := client.Search(ctx, request)
+		require.NoError(err)
+		require.Empty(reply.Error)
+		require.Len(reply.Results, 1)
+
+		require.Equal(hotelVASP.Id, reply.Results[0].Id)
+		require.Equal(hotelVASP.RegisteredDirectory, reply.Results[0].RegisteredDirectory)
+		require.Equal(hotelVASP.CommonName, reply.Results[0].CommonName)
+		require.Equal(hotelVASP.TrisaEndpoint, reply.Results[0].Endpoint)
+	})
+
+	s.Run("Verified", func() {
+		// Only verified results are returned
+		require := s.Require()
+		request := &api.SearchRequest{
+			Name: []string{"CharlieBank", "Hotel Corp"},
+		}
 
-	// Filter by business category - no results
-	request = &api.SearchRequest{
-		Name:             []string{"CharlieBank"},
-		BusinessCategory: []pb.BusinessCategory{pb.BusinessCategory_GOVERNMENT_ENTITY},
-	}
-	reply, err = client.Search(ctx, request)
-	require.NoError(err)
-	require.Empty(reply.Error)
-	require.Len(reply.Results, 0)
+		reply, err := client.Search(ctx, request)
+		require.NoError(err)
+		require.Empty(reply.Error)
+		require.Len(reply.Results, 1)
+		require.Equal(hotelVASP.Id, reply.Results[0].Id)
+	})
+
+	s.Run("FuzzyPrefix", func() {
+		// Fuzzy search by case-insensitive prefix
+		require := s.Require()
+		request := &api.SearchRequest{
+			Name: []string{"NOV"},
+		}
 
-	// Filter by VASP category
-	request = &api.SearchRequest{
-		Name:         []string{"CharlieBank"},
-		VaspCategory: []string{"P2P"},
-	}
-	reply, err = client.Search(ctx, request)
-	require.NoError(err)
-	require.Empty(reply.Error)
-	require.Len(reply.Results, 1)
+		reply, err := client.Search(ctx, request)
+		require.NoError(err)
+		require.Empty(reply.Error)
+		require.Len(reply.Results, 1)
+		require.Equal(novemberVASP.Id, reply.Results[0].Id)
+
+		// Prefix search must have at least three characters
```

Comment:
Is this behavior documented on the Search endpoint?

Reply:
That's a good question -- the documentation doesn't mention it. Might be something to comment on.